src/base/bittorrent/session.cpp

```diff
@@ -321,6 +321,11 @@ QString Session::tempPath() const
     return m_tempPath;
 }
 
+QString Session::torrentTempPath(const InfoHash &hash) const
+{
+    return m_tempPath + QString(hash) + "/";
```

Comment:
See https://en.wikipedia.org/wiki/Comparison_of_file_systems#Limits
IIRC most modern filesystems has no limit for path names. Names can be limited by the driver or by the operating system, but it's enough to not worry about the extra 40 characters.